controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -252,7 +263,16 @@ func (r *OCSInitializationReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			return strings.HasPrefix(client.GetName(), PrometheusOperatorCSVNamePrefix)
 		},
 	)
-	return ctrl.NewControllerManagedBy(mgr).
+
+	clusterClaimCrdPredicate := predicate.Funcs{
+		CreateFunc: func(e event.TypedCreateEvent[client.Object]) bool {
+			panic("ClusterClaim CRD was found. Restarting pod to initiate creation")
+		},
+		DeleteFunc: func(e event.TypedDeleteEvent[client.Object]) bool {
+			panic("ClusterClaim CRD was found. Restarting pod to initiate deletion")
+		},
+	}
```

Comment:
I'd like to avoid setting panic to restart the process as much as possible. There are ways to set watches dynamically, we should try those first.

Reply:
@umangachapagain They don't work because they do not re-popolate the cache. The only reliable way we found was to panic out. 

@raaizik I would not use panic, I will use `os.exit` with a specific code. That way we can warp the operator invocation with a small bash command that invokes the operator process if it sees that particular exit code. 
Doing it that way will prevent unnecessary pos restarts